src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -434,6 +434,71 @@ public void TestCreatingActivityUsingDifferentParentIds()
             }).Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestActivityContextIsRemote()
+        {
+            RemoteExecutor.Invoke(() => {
+                ActivityContext ctx = new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), default);
+                Assert.False(ctx.IsRemote);
+
+                bool isRemote = false;
+
+                using ActivitySource aSource = new ActivitySource("RemoteContext");
+                using ActivityListener listener = new ActivityListener();
+                listener.ShouldListenTo = (activitySource) => activitySource.Name == "RemoteContext";
+
+                listener.GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) =>
+                {
+                    isRemote = activityOptions.Parent.IsRemote;
+                    return ActivityDataRequest.AllData;
+                };
+
+                ActivitySource.AddActivityListener(listener);
+
+                foreach (bool b in new bool[] { true, false })
+                {
+                    ctx = new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), default, default, b);
+                    Assert.Equal(b, ctx.IsRemote);
+
+                    aSource.StartActivity("a1", default, ctx);
+                    Assert.Equal(b , isRemote);
+                }
+            }).Dispose();
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestTraceIdAutoGeneration()
+        {
+            RemoteExecutor.Invoke(() => {
+
+                using ActivitySource aSource = new ActivitySource("TraceIdAutoGeneration");
+                using ActivityListener listener = new ActivityListener();
+                listener.ShouldListenTo = (activitySource) => activitySource.Name == "TraceIdAutoGeneration";
+
+                ActivityContext ctx = default;
+
+                listener.GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) =>
+                {
+                    ctx = activityOptions.Parent;
+                    return ActivityDataRequest.AllData;
+                };
+
+                ActivitySource.AddActivityListener(listener);
+
+                aSource.StartActivity("a1", default, ctx);
+                Assert.Equal(default, ctx);
+
+                listener.AutoGenerateRootContextTraceId = true;
+
+                Activity activity = aSource.StartActivity("a2", default, ctx);
```

Comment:
`aSource.StartActivity("a2", default, null)` - could we test with this as well? This should trigger `GetRequestedDataUsingParentId`, which would in turn would call `GetRequestedDataUsingContext`, with auto-generated traceid. (from an earlier PR which added this behavior)

Reply:
I am not sure I understand the ask here. but anyway, we can add any suggested tests in other PR as I want to get this soon.

Reply:
I added this test too.